dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/LDAPMemberAttributeScopeTest.java

```diff
@@ -0,0 +1,577 @@
+        results = urServlet.getGroupsForUser(MEMBER_DYNAMIC_USER_DN);
+        assertMembershipResults("With nested scope, the dynamic member should be a member of the dynamic group.", results, MEMBER_LEVEL1_GROUP_DN, MEMBER_DYNAMIC_GROUP_DN);
+
+        // TODO user_level1 is returned b/c recursiveSearch=true overrides search scope in LdapAdapter.getMembersByMember()
+        results = urServlet.getUsersForGroup(MEMBER_LEVEL1_GROUP_DN, 0).getList();
+        assertMembershipResults("With nested scope, the root group should have the direct member.", results, MEMBER_LEVEL1_USER_DN, MEMBER_LEVEL2_USER_DN);
+
+        results = urServlet.getUsersForGroup(MEMBER_LEVEL2_GROUP_DN, 0).getList();
+        assertMembershipResults("With nested scope, the nested group should have only the nested member.", results, MEMBER_LEVEL2_USER_DN);
+
+        results = urServlet.getUsersForGroup(MEMBER_DYNAMIC_GROUP_DN, 0).getList();
+        assertMembershipResults("With nested scope, the dynamic group should have only the dynamic member.", results, MEMBER_DYNAMIC_USER_DN);
+    }
+
+    /**
+     * Test a memberAttribute whose scope is set to "all". All scoped member attributes
+     * return their direct, nested and dynamic members, thus this will NOT require nested calls
+     * to derive group membership.
+     *
+     * @throws Exception if the test failed for some unforeseen reason.
+     */
+    @Test
+    public void memberAttribute_scope_all() throws Exception {
+
+        ServerConfiguration clone = baseConfiguration.clone();
+        MemberAttribute[] memberAttrs = new MemberAttribute[1];
+        memberAttrs[0] = new MemberAttribute(null, "member", "groupofnames", LdapConstants.LDAP_ALL_GROUP_MEMBERSHIP_STRING);
+        createLdapRegistry(clone, MEMBER_BASE_DN, memberAttrs, null);
+        LDAPFatUtils.updateConfigDynamically(libertyServer, clone);
+
+        List<String> results = urServlet.getGroupsForUser(MEMBER_LEVEL1_USER_DN);
+        assertMembershipResults("With all scope, the direct member should only be a member of the root group.", results, MEMBER_LEVEL1_GROUP_DN);
+
+        // TODO group_level1 is returned b/c recursiveSearch=true overrides search scope in LdapAdapter.getGroupsByMember()
+        results = urServlet.getGroupsForUser(MEMBER_LEVEL2_USER_DN);
+        assertMembershipResults("With all scope, the nested member should only be a member of the nested group.", results, MEMBER_LEVEL1_GROUP_DN, MEMBER_LEVEL2_GROUP_DN);
+
+        // TODO group_level1 is returned b/c recursiveSearch=true overrides search scope in LdapAdapter.getGroupsByMember()
+        results = urServlet.getGroupsForUser(MEMBER_DYNAMIC_USER_DN);
+        assertMembershipResults("With all scope, the dynamic member should be a member of the dynamic group.", results, MEMBER_LEVEL1_GROUP_DN, MEMBER_DYNAMIC_GROUP_DN);
+
+        // TODO user_level1 is returned b/c recursiveSearch=true overrides search scope in LdapAdapter.getMembersByMember()
```

Comment:
The dynamic user should be a member of the level 1 group, but the LdapAdapter.getMembersByMembership() method does not consider nested dynamic groups.